### DISCUSSION TITLE
the URL must start with the protocol `postgresql://` or `postgres://`, whilst using Prisma Accelerate.

### DISCUSSION THREAD

**Discussion by tobyab** [prisma/prisma#23448] [Q&A] [answered]
### Question

Hey! I'm using Prisma accelerate in my Prisma application. I'm using a database url with a `prisma://` protocol. However, I'm getting the error:
```
This request could not be understood by the server: {"type":"UnknownJsonError","body":{"code":"P6008","message":"Accelerate was not able to connect to your database. The underlying error is: Error validating datasource `db`: the URL must start with the protocol `postgresql://` or `postgres://`."}} (The request id was: 862e9f38aa4471d8)
```
I'd love some help on this. Thank you so much!

### How to reproduce (optional)

"@prisma/extension-accelerate": "^1.0.0",
"prisma": "5.10.2",
"@prisma/client": "5.10.2",

### Information about Prisma Schema, Client Queries and Environment (optional)

```
lib/prisma.ts
---

import { PrismaClient } from "@prisma/client/edge";
import { withAccelerate } from "@prisma/extension-accelerate";

const prisma = new PrismaClient().$extends(withAccelerate());

export default prisma;
```

```
prisma/schema.prisma
---

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider     = "postgresql"
  url          = env("DATABASE_URL")
  directUrl    = env("DIRECT_DATABASE_URL")
  relationMode = "prisma"
}

model User {
  id        String    @id @default(cuid())
  createdAt DateTime? @default(now()) @db.Timestamptz(6)
  updatedAt DateTime? @db.Timestamptz(6)
  username  String?   @unique
  name      String?
  suspended Boolean   @default(false)
  admin     Boolean   @default(false)
  email     String?   @unique
  image     String?
}
```

```
.env
---

DATABASE_URL="prisma://accelerate.prisma-data.net/?api_key=<api_key>"
DIRECT_DATABASE_URL="postgresql://example:example2@super-cool-name.us-west-2.aws.neon.tech/groovy?sslmode=require"
```

```
app/api/example.route.ts
---
export async function POST() {
	await prisma.user.create({
		data: {
                  createdAt: new Date().toISOString(),
                  updatedAt: new Date().toISOString(),
                  username: "toby"
		},
	});
}
```

**Comment by janpio** [comment, marked as answer]
What connection string did you configure in Prisma Accelerate?

To me it seems that this error message is from Accelerate on the cloud side, not from Prisma ORM in your application. So Prisma ORM and Client are probably very happy with the `prisma://` connection string you provided, but Accelerate itself is having trouble.

**Reply by tobyab** [reply]
Ah, yes! Of course. The connection string on Accelerate's side is out of date. Thank you so much for your help, mate!

**Reply by mavilein** [reply]
We had a discussion internally about how the error message is not super helpful in this case. We think we can improve this error message in the future. To give a bit of background: 6008 is our generic error for when the connection to the database can not be established. We will keep it that way but could massage the message a bit to improve this specific situation.

On another note: You are using an outdated version of the Accelerate extension or even not at all. This is indicated by the presence of that `UnknownJsonError`. With a more recent extension this should look a little nicer already.
